src/core/continuous_space.jl

```diff
@@ -350,7 +368,20 @@ function interacting_pairs(model, r, scheduler = model.scheduler)
       push!(pairs, (id, a2.id))
     end
   end
-  return PairIterator(pairs, model.agents)
+end
+
+function all_pairs!(pairs::Vector{Tuple{Int, Int}}, model::ABM, r)
+  for id in keys(model.agents)
+    for nid in space_neighbors(model[id], model, r)
+      # Sort the pair to overcome any uniqueness issues
+      new_pair = isless(id, nid) ? (id, nid) : (nid, id)
+      !(new_pair in pairs) && push!(pairs, new_pair)

```

Comment:
Since you sort the pairs anyway, it might be more performant to use commands like `searchsortedfirst` instead of `in` and check whether `searchsortedfirst` returns length()+1 or something less.

Reply:
Ah no, what I said is incorrect, ou don't sort the pairs, you only sort the ID within the pairs. Sorting the outer vector as well would take too much time.

Reply:
OK, I'll look into that.

Reply:
no reason!

Reply:
Yeah exactly. Wanted to avoid that.